extensions/wikia/FacebookTags/FacebookTagParser.php

```diff
@@ -14,15 +14,32 @@ public function __construct( string $tagName ) {
 	}
 
 	public function parse( $content, array $attributes, Parser $parser, PPFrame $frame ): string {
-		$pluginOptionWhitelist = FacebookTagConstants::SUPPORTED_TAG_ATTRIBUTES +
-								 FacebookTagConstants::FB_PLUGIN_OPTS[$this->tagName];
-		$tagAttributes = Sanitizer::validateAttributes( $attributes, $pluginOptionWhitelist );
+		$tagAttributes = $this->buildTagAttributes( $attributes );
+
+		return Html::element( 'iframe', $tagAttributes );
+	}
+
+	private function buildTagAttributes( array $attributes ): array {
+		$attributeWhitelist = FacebookTagConstants::SUPPORTED_TAG_ATTRIBUTES;
+		$tagAttributes = Sanitizer::validateAttributes( $attributes, $attributeWhitelist );
+
+		$queryParameters = $this->buildQueryParameters( $attributes );
+		$fbUrl = sprintf( FacebookTagConstants::FB_API_URL, $this->tagName, $queryParameters );
 
-		// used as selector by JS to determine if there are any widgets to render on the page
 		$tagAttributes += [
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$tagAttributes". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)